### DISCUSSION TITLE
Uncaught TypeError: Designer is not a constructor

### DISCUSSION THREAD

**Discussion by ryo-ma1010** [pdfme/pdfme#409] [Q&A]
Hello I am trying out pdfme for the first time at https://pdfme.com/docs/getting-started. 
I am creating a ts file in this section https://pdfme.com/docs/getting-started#designer.
But I got the following error and I could not get the template designer to work. What should I do to fix this error?

```
pdfme.js:29 Uncaught TypeError: Designer is not a constructor
```

Here is the ts file I created.

```
import { Template, BLANK_PDF } from '@pdfme/common';
import { Designer } from '@pdfme/ui';

const domContainer = document.getElementById('container')!;
const template: Template = {
    basePdf: BLANK_PDF,
    schemas: [
      {
        a: {
          type: 'text',
          position: { x: 0, y: 0 },
          width: 10,
          height: 10,
        },
        b: {
          type: 'text',
          position: { x: 10, y: 10 },
          width: 10,
          height: 10,
        },
        c: {
          type: 'text',
          position: { x: 20, y: 20 },
          width: 10,
          height: 10,
        },
      },
    ],
};
const desiner = new Designer({ domContainer, template }); // Uncaught TypeError: Designer is not a constructor

```

I look forward to hearing your opinions or advice.
Thank you in advance.

**Comment by hand-dot** [comment]
Can you tell me a reproducable repo?

**Reply by hand-dot** [reply]
Your code looks good. But I don't know your environment

**Reply by ryo-ma1010** [reply]
Thank you for your reply.
Here is the repository.
https://github.com/ryo-ma1010/MyLaravel

I have installed Pdfme on Laravel.
The environment is as follows.

```
Laravel 10.41
PHP 8.3
vite 5.0.0
nodejs 20.11
react 18.2
```

Please run `npm run dev` and access `localhost`.
The template for the top page is `resources/views/welcome.blade.php`.
I appreciate your help.

**Reply by hand-dot** [reply]
Sorry I can't run your app on my machine. I don't have docker.

I guess this is a build or call of code problem. this is not pdfme problem.
because our website uses Designer as a constructor. 

https://github.com/pdfme/pdfme/blob/85ddf3ea67c4c631e8387c2d497288ed293c747c/website/src/pages/template-design.tsx#L63

**Reply by hand-dot** [reply]
Hey @ryo-ma1010 ,

There was a report from a user in the pdfme Discord who is facing the same issue.
https://discord.com/channels/933602537808990268/933602537808990271/1199906876792524840

I was able to reproduce it with a minimal setup of Vite, so I want to fix it to prevent the same error from occurring.

I will create an issue later and fix it over the weekend if I have time.

**Reply by hand-dot** [reply]
New issue: https://github.com/pdfme/pdfme/issues/413

so I gonna close this thread and will be handle in the above issue.
